File: src/main/java/fr/lewon/web/bot/entities/enums/Rarity.java
package fr.lewon.web.bot.entities.enums;

public enum Rarity {

	COMMON("common", 1),
	RARE("rare", 2),
	EPIC("epic", 3),
	LEGENDARY("legendary", 4);

	private final String label;
	private final Integer value;

	private Rarity(String label, Integer value) {
		this.label = label;
		this.value = value;
	}

	public String getLabel() {
		return label;
	}

	public Integer getValue() {
		return value;
	}

	public static Rarity getValueByLabel(String label) {
		for (Rarity r : values()) {
			if (r.getLabel().equals(label)) {
				return r;
			}
		}
		return null;
	}
}

File: src/main/kotlin/fr/lewon/bot/hh/entities/enums/Rarity.kt
package fr.lewon.bot.hh.entities.enums

enum class Rarity(val label: String, val value: Int) {
    STARTING("starting", 0),
    COMMON("common", 1),
    RARE("rare", 2),
    EPIC("epic", 3),
    LEGENDARY("legendary", 4);

    companion object {
        fun getValueByLabel(label: String?): Rarity? {
            for (r in values()) {
                if (r.label == label) {
                    return r
                }
            }
            return null
        }
    }

}